tests/cache/sampleStoreFlagFlipFlop.js

```diff
@@ -0,0 +1,235 @@
+  let s3;
+
+  before((done) => {
+    tu.toggleOverride(sampleStore.constants.featureName, false);
+    Aspect.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Aspect1`,
+      timeout: '30s',
+      valueType: 'NUMERIC',
+      criticalRange: [0, 1],
+      relatedLinks: [
+        { name: 'Google', value: 'http://www.google.com' },
+        { name: 'Yahoo', value: 'http://www.yahoo.com' },
+      ],
+    })
+    .then((created) => (a1 = created))
+    .then(() => Aspect.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Aspect2`,
+      timeout: '10m',
+      valueType: 'BOOLEAN',
+      okRange: [10, 100],
+    }))
+    .then((created) => (a2 = created))
+    .then(() => Aspect.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Aspect3`,
+      timeout: '10m',
+      valueType: 'BOOLEAN',
+      okRange: [10, 100],
+    }))
+    .then((created) => (a3 = created))
+    .then(() => Aspect.create({
+      isPublished: false,
+      name: `${tu.namePrefix}Aspect4`,
+      timeout: '10m',
+      valueType: 'BOOLEAN',
+      okRange: [10, 100],
+    }))
+    .then((created) => (a4 = created))
+    .then(() => Subject.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Subject1`,
+    }))
+    .then((created) => (s1 = created))
+    .then(() => Subject.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Subject2`,
+      parentId: s1.id,
+    }))
+    .then((created) => (s2 = created))
+    .then(() => Subject.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Subject3`,
+      parentId: s1.id,
+    }))
+    .then((created) => (s3 = created))
+    .then(() => Sample.create({
+      subjectId: s2.id,
+      aspectId: a1.id,
+      value: '0',
+      relatedLinks: [
+        { name: 'Salesforce', value: 'http://www.salesforce.com' },
+      ]
+    }))
+    .then(() => Sample.create({
+      subjectId: s2.id,
+      aspectId: a2.id,
+      value: '50',
+      relatedLinks: [
+        { name: 'Salesforce', value: 'http://www.salesforce.com' },
+      ]
+    }))
+    .then(() => Sample.create({
+      subjectId: s3.id,
+      aspectId: a1.id,
+      value: '5',
+      relatedLinks: [
+        { name: 'Salesforce', value: 'http://www.salesforce.com' },
+      ]
+    }))
+    .then(() => done())
+    .catch(done);
+  });
+
+  after((done) => {
+    u.forceDelete(done)
+    .then(() => redisClient.flushallAsync())
+    .then(() => tu.toggleOverride(sampleStore.constants.featureName,
+      initialFeatureState))
+    .then(() => done())
+    .catch(done);
+  });
+
+  it('flag from false to false: cache should not be populated', (done) => {
+    samstoinit.eradicate()
+    .then(() => tu.toggleOverride(sampleStore.constants.featureName, false))
+    .then(() => samstoinit.init())
+    .then((res) => expect(res).to.be.false)
+    .then(() =>
+      redisClient.smembersAsync(sampleStore.constants.indexKey.aspect))
+    .then((res) => {
+      expect(res).to.have.length(0);
```

Comment:
you can return the promise here, instead of doing another .then
return  redisClient.smembersAsync(sampleStore.constants.indexKey.sample))
})
.then((res) => {
expect...
return ...
})